core/rtmp/handler.go

```diff
@@ -76,8 +76,8 @@ func (h *Handler) OnPublish(timestamp uint32, cmd *rtmpmsg.NetStreamPublish) err
 	}
 	h.flvEnc = enc
 
-	//TODO: why is this turned into a goroutine?
-	go ffmpeg.Start()
+	transcoder := ffmpeg.CreateTranscoderFromConfig()
+	go transcoder.Start()
```

Comment:
This is run under a Goroutine because the execution is blocking, `ffmpeg` will keep running.  Let me know if I'm missing something or you have some other suggestions!